File: Solved/0705/0705.cpp
// Problem No.: 705
// Solver:      Jinmin Goh
// Date:        20200803
// URL: https://leetcode.com/problems/design-hashset/

#include<set>
#include<iterator>

using namespace std;

class MyHashSet {
public:
    set<int> hashSet;
    MyHashSet() {
        //set<int> hashSet;
    }
    
    void add(int key) {
        if(hashSet.find(key) == hashSet.end()) {
            hashSet.insert(hashSet.begin(), key);
        }
    }
    
    void remove(int key) {
        if(hashSet.find(key) != hashSet.end()) {
            hashSet.erase(hashSet.find(key));
        }
    }
    
    /** Returns true if this set contains the specified element */
    bool contains(int key) {
        if(hashSet.find(key) == hashSet.end()) {
            return false;
        }
        else {
            return true;
        }
    }
};
            

File: Solved/0705/0705.py
# Problem No.: 705
# Solver:      Jinmin Goh
# Date:        20200803
# URL: https://leetcode.com/problems/design-hashset/

import sys

class MyHashSet:

    def __init__(self):
        self.hashSet = set()
        

    def add(self, key: int) -> None:
        self.hashSet.add(key)

    def remove(self, key: int) -> None:
        if key in self.hashSet:
            self.hashSet.remove(key)

    def contains(self, key: int) -> bool:
        if key in self.hashSet:
            return True
        else:
            return False
            